### DISCUSSION TITLE
End to End tests

### DISCUSSION THREAD

**Discussion by julianocosta89** [open-telemetry/opentelemetry-demo#199] [Ideas]
Hello all 👋🏽 

@mic-max is working on tests to ensure new changes do not break the demo-webstore.
Thanks for that 🤩 !

I still believe we need to think in a different way of putting the tests together though.
Ensuring that the service is running and generating Traces is important, but it doesn't ensure that the spans are linked together, and it could happen we would have broken traces with orphaned spans.

I'm opening a discussion because ATM I do not have an easier solution, but maybe someone involved in the project has.

**Comment by mathnogueira** [comment]
We can try building a PoC with [Tracetest](https://github.com/kubeshop/tracetest) to test the generated traces. We can keep it in a different repo for now.

**Comment by julianocosta89** [comment]
This was solved some time ago and now every time we have an approval from approvers or maintainers in the PRs, the TraceTest is executed, ensuring the traces are not broken. 🎉
